### PR TITLE
docs/geojson

### LINKED ISSUE #13651: chart.map type
#### Expected behaviour
`chart.map` property should accept JSON imported from official map collection ('@highcharts/map-collection/custom/world.geo.json').

#### Actual behaviour
```typescript
Type 'typeof import("*.json")' is not assignable to type 'string | any[]'.
Type 'typeof import("*.json")' is missing the following properties from type 'any[]': length, pop, push, concat, and 25 more.
```

#### Live demo with steps to reproduce
https://stackblitz.com/edit/highcharts-angular-map-drjs4y

#### Workaround
cast imported map to the `any`

#### Product version
8.1.0

#### Affected browser(s)
All

### PR DESCRIPTION
**Pull request by bre1470**
Closed #13651, Added interface for geo JSON files.

### CODE REVIEW

File: ts/parts-map/GeoJSON.ts [hunk 1]
```diff
@@ -71,9 +71,23 @@ declare global {
             /** @requires modules/map */
             proj4?: any;
         }
+        interface GeoJSON {
+            copyright?: string;
+            copyrightShort?: string;
+            crs?: Record<string, any>;
+            features: Array<Record<string, any>>;
+            'hc-transform'?: Record<string, GeoJSONFeature>;
```
[the remark this answers is not in the document]
Comment (KacperMadej) [reply ts/parts-map/GeoJSON.ts:79]: Additionally to the code comment:
If I change definitions to
```
          features: Array<GeoJSONFeature>;
          'hc-transform'?: Record<string, any>;
```
(and test in an Angular app with `tsconfig.json` having `"resolveJsonModule": true,` in `compilerOptions`)
then there is no error.

[the remark this answers is not in the document]
Comment (bre1470) [reply ts/parts-map/GeoJSON.ts:79]: Hm, that does not sound right. You should not use the internal namespace in an angular project. It serves only the implementation code but not much the usage. The declarations in `./code` are the ones for users. To generate, use `npx gulp scripts--force && npx gulp jsdoc-dts`.

**Comment by bre1470** [comment]
Thanks, @KacperMadej ! Now I understand the problem.

One question is left though. The `crs` option can be in GeoJSON and in HC transform? Or should it be only in HC tranform? @oysteinmoseng ?

File: ts/parts-map/GeoJSON.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bre1470) [reply ts/parts-map/GeoJSON.ts:79]: Now I got that, thanks!

**Comment by oysteinmoseng** [comment]
`crs` can be in either place. Any reason why we could not add the full definitions for `hc-transform`? The object is defined by us (see article linked above).

**Comment by bre1470** [comment]
I can do the internal interface, but for the doclets I am lacking the insight to explain the usage of the options.

**Comment by oysteinmoseng** [comment]
No worries, I can add some text to the doclets if you put in placeholders and create the interface

**Comment by bre1470** [comment]
@oysteinmoseng Thank you! 🎉

**Pull request merged**